libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -140,18 +136,26 @@ void ControlFlowAnalyzer::checkUninitializedAccess(CFGNode const* _entry, CFGNod
 				ssl.append("The variable was declared here.", variableOccurrence->declaration().location());
 
 			bool isStorage = variableOccurrence->declaration().type()->dataStoredIn(DataLocation::Storage);
-			m_errorReporter.typeError(
-				3464_error,
-				variableOccurrence->occurrence() ?
-					*variableOccurrence->occurrence() :
+			bool isCalldata = variableOccurrence->declaration().type()->dataStoredIn(DataLocation::CallData);
+			if (isStorage || isCalldata)
+				m_errorReporter.typeError(
+					3464_error,
+					variableOccurrence->occurrence() ?
+						*variableOccurrence->occurrence() :
+						variableOccurrence->declaration().location(),
+					ssl,
+					"This variable is of " +
+					string(isStorage ? "storage" : "calldata") +
+					" pointer type and can be " +
+					(variableOccurrence->kind() == VariableOccurrence::Kind::Return ? "returned" : "accessed") +
+					" without prior assignment, which would lead to undefined behaviour."
+				);
+			else if (!_emptyBody && variableOccurrence->declaration().name().empty())
+				m_errorReporter.warning(
+					6321_error,
 					variableOccurrence->declaration().location(),
-				ssl,
-				"This variable is of " +
-				string(isStorage ? "storage" : "calldata") +
-				" pointer type and can be " +
-				(variableOccurrence->kind() == VariableOccurrence::Kind::Return ? "returned" : "accessed") +
-				" without prior assignment, which would lead to undefined behaviour."
-			);
+					"Unnamed return parameter can remain uninitialized."
```

Comment:
I would modify the wording here a bit - because it does not remain uninitialized.
```suggestion
					"Unnamed return parameter can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the parameter."
```